lib/open_mic.rb

```diff
@@ -0,0 +1,34 @@
+class OpenMic
+  attr_reader :location,
+              :date,
+              :performers
+
+  def initialize(data_hash)
+    @location   = data_hash[:location]
+    @date       = data_hash[:date]
+    @performers = []
+  end
+
+  def welcome(user)
+    @performers.push(user)
+  end
+
+  def repeated_jokes?
```

Comment:
I think you may have over-engineered this just a bit!  It looks like your goal is to go through each performer and check all of their jokes against another performers jokes.  This will work, but you could think about it slightly differently with the same result: 

If you get a list of all jokes known by performers, you can compare a unique list of those jokes against the full list to see if there are any duplicates (this has the added benefit of testing uniqueness of jokes even within a single comedian) It could look something like this:

```ruby
def repeated_jokes?
  all_jokes = @performers.map do |performer|
    performer.jokes
  end.flatten

  all_jokes.uniq < all_jokes
end
```